pod/pod.go

```diff
@@ -74,3 +74,39 @@ func (s *Service) Delete(ctx context.Context, name string) error {
 
 	return nil
 }
+
+// DelteAll deletes all pods.
+func (s *Service) DeleteAll(ctx context.Context) error {
```

Comment:
In this DeleteAll method, you should delete all Pods.
But, in DeleteAll of node service, you shouldn't delete all Pods but delete all scheduled Pods.

Reply:
So,
- rename this method to `DeleteAllScheduledPod`
- use `DeleteAllScheduledPod` from `DeleteAll` in node service
- create new `DeleteAll` which deletes all pods(= scheduled pods + unscheduled pods.)

Reply:
Thanks for review.

> create new DeleteAll which deletes all pods(= scheduled pods + unscheduled pods.)

This method will be not used now, so I think it's better not to create it.
What do you think ?

Reply:
> This method will be not used now, 

No. `DeleteAll` method is needed to delete all Pods.

You'd like to create "delete all" method in each service now, right?

Deleting all Nodes means "delete all Nodes and delete all scheduled Pods". So, you need to create "DeleteAllScheduledPod" to delete all scheduled Pods.

Deleting all Pods means "delete all pods whether they are scheduled or not." So, you also need to create "DeleteAll" to delete all Pods.

Reply:
Ah, this PR is scoped to create `delete all` method in each service, so we should create `DeleteAllScheduledPod` and `DeleteAll`, right ?

> Deleting all Nodes means "delete all Nodes and delete all scheduled Pods". So, you need to create 
> "DeleteAllScheduledPod" to delete all scheduled Pods in "DeleteAll" in node service.

> Deleting all Pods means "delete all pods whether they are scheduled or not." So, you also need to create 
> "DeleteAll" to delete all Pods.

I know, thanks.

Reply:
@sanposhiho 

I added DeleteAll method, too. `pod service` test isn't existed, now.
Shall I add a test? Or do you hope to create different PR because the scope is larger than this PR?

Reply:
Oh, I've just commented about the test. Yes, please add test for DeleteAllScheduledPod.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/68#discussion_r780659785